### PR TITLE
EclGenericVanguard: add serialization for dynamic state

### PR DESCRIPTION
**Pull request by akva2**
Sits on top of https://github.com/OPM/opm-simulators/pull/4419
Sits on top of https://github.com/OPM/opm-simulators/pull/4417

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclgenericvanguard.hh [hunk 1]
```diff
@@ -84,13 +87,16 @@ public:
      * \details Needs to be in compile unit.
      */
     EclGenericVanguard();
+    EclGenericVanguard(SimulationModelParams&& params);
```
Comment (bska) [inline ebos/eclgenericvanguard.hh:90]: Single-parameter constructor.  Is there a reason for not marking this `explicit`?

File: tests/test_RestartSerialization.cpp [hunk 2]
```diff
@@ -75,6 +81,28 @@ TEST_FOR_TYPE_NAMED_OBJ(ATE, AdaptiveTimeSteppingEbosPID, serializationTestObjec
...
 TEST_FOR_TYPE_NAMED_OBJ(ATE, AdaptiveTimeSteppingEbosSimple, serializationTestObjectSimple)
 
+BOOST_AUTO_TEST_CASE(EclGenericVanguard)
+{
+    auto in_params = Opm::EclGenericVanguard::serializationTestParams();
+    Opm::EclGenericVanguard val1(std::move(in_params));
+    Opm::Serialization::MemPacker packer;
+    Opm::Serializer ser(packer);
+    ser.pack(val1);
+    size_t pos1 = ser.position();
```
Comment (bska) [inline tests/test_RestartSerialization.cpp:91]: It probably doesn't matter a whole lot, but I think `pos1` (and `pos2` below) could be marked `const`.

**Review by bska — COMMENTED** [review]
Looks good to me for the most part.  Just a couple of nits.

File: ebos/eclgenericvanguard.hh (hunk 1 above)
Comment (akva2) [reply ebos/eclgenericvanguard.hh]: no reason, fixed.

File: tests/test_RestartSerialization.cpp (hunk 2 above)
Comment (akva2) [reply tests/test_RestartSerialization.cpp]: sure.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge into master.

**Pull request merged**